src/NuGet.Core/NuGet.Commands/RestoreCommand/FrameworkRuntimeDefinition.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Globalization;
+using NuGet.Frameworks;
+
+namespace NuGet.Commands
+{
+    internal sealed record FrameworkRuntimeDefinition
+    {
+        public string TargetAlias { get; }
+
+        public NuGetFramework Framework { get; }
+
+        public string RuntimeIdentifier { get; }
+
+        public string Name { get; }
+
+        public FrameworkRuntimeDefinition(string targetAlias, NuGetFramework framework, string? runtimeIdentifier)
+        {
+            TargetAlias = targetAlias ?? string.Empty;
+            Framework = framework ?? throw new ArgumentNullException(nameof(framework));
+            RuntimeIdentifier = runtimeIdentifier ?? string.Empty;
+            Name = FrameworkRuntimePair.GetTargetGraphName(framework, runtimeIdentifier);
```

Comment:
This name is used for logging, should it also have the alias now?  Otherwise you might have set an alias for a two of the same framework but the error will only have the framework and not the alias a user specified right?

Reply:
That's the idea, but not yet in this PR.

That requires a lot more changes, so I'm trying to separate between the refactoring that we have here and functional changes such as changing up the error messages. 

Basically I need to make the changes in a lot more places where we raise warnings.